test/e2e/framework/pod/resource.go

```diff
@@ -388,6 +393,41 @@ func logPodTerminationMessages(pods []v1.Pod) {
 	}
 }
 
+// logPodLogs logs the container logs from pods in the given namespace. This can be helpful for debugging
+// issues that do not cause the container to fail (e.g.: network connectivity issues)
+// We will log the Pods that have the LabelLogOnPodFailure label. If there aren't any, we default to
+// logging only the first 5 Pods.
+func logPodLogs(c clientset.Interface, namespace string, pods []v1.Pod) {
+	var logPods []v1.Pod
+	for _, pod := range pods {
+		if _, ok := pod.Labels[LabelLogOnPodFailure]; ok {
+			logPods = append(logPods, pod)
+		}
+	}
+	maxPods := len(logPods)
+
+	// There are no pods with the LabelLogOnPodFailure label, we default to the first 5 Pods.
+	if maxPods == 0 {
+		logPods = pods
+		maxPods = len(pods)
+		if maxPods > 5 {
+			maxPods = 5
+		}
+	}
+
+	for i := 0; i < maxPods; i++ {
+		pod := logPods[i]
+		for _, container := range pod.Spec.Containers {
+			logs, err := getPodLogsInternal(c, namespace, pod.Name, container.Name, false, nil, 42)
+			if err != nil {
+				e2elog.Logf("unable to fetch %s/%s/%s logs: %v", pod.Namespace, pod.Name, container.Name, err)
+				continue
+			}
+			e2elog.Logf("%s/%s/%s logs:\n%s", pod.Namespace, pod.Name, container.Name, logs)
```

Comment:
I'm not sure how I feel about polluting build-log.txt with this.  What about dumping into something like `artifacts/pods/{namespace}/{pod}/{container}/logs.txt`?

Reply:
Done